### ISSUE TITLE
Build snippet with embedded email opt in fields

### ISSUE THREAD

**Issue by valianttry** [MozillaFoundation/encryption_campaign#23]
see #22 

Opt ins should land in Exact Target

two versions:
1) with video replacing the Firefox logo; and 2) one with video placed below the logo and search bar.

rough mockup for above search bar video placement:
![snippet_encryption_campaign_-_google_slides](https://cloud.githubusercontent.com/assets/9830371/12317279/8f25b636-ba44-11e5-9d93-aa8f5866c216.png)

Below search bar:

![snippet_encryption_campaign_-_google_slides](https://cloud.githubusercontent.com/assets/9830371/12317350/519845bc-ba45-11e5-9914-3a910ce3a315.png)

**Comment by valianttry** [comment]
We should probably build a version with no embedded video as well. Just in case. :8ball:

**Comment by valianttry** [comment]
Also we should be able to test removing some of the fields, such as name.

**Comment by OpenMatt** [comment]
@ScottDowne does this all make sense to you?

**Comment by ScottDowne** [comment]
Yup sounds good.

**Comment by ScottDowne** [comment]
Just a heads up, we've previously operated under the limitation that exact target cannot store names, just email, lang, and country.

Regular mailing list signups go to exact target, without storing a name, and petitions have been going to BSD where we can and need to get a name.

So, we're currently left we just email and country for signups.

**Comment by valianttry** [comment]
@adamlofting there's no way to add name?

**Comment by ScottDowne** [comment]
@valianttry Do you think we need a privacy checkbox on this?

**Comment by valianttry** [comment]
I'm going to assume there is no need for an opt-in checkbox, because there is only 1 task we're asking the user to perform. We will also be sending them an instant auto-email to confirm their subscription.

We do need to include disclaimer language: 
"By clicking "Submit" you will be subscribed to email updates and are okay with Mozilla handling your data as explained in **this Privacy Policy**." 

this Privacy Policy should be linked to: https://www.mozilla.org/privacy/websites/

@ScottDowne is it difficult to make the disclaimer language editable just in case legal has tweaks?

Additional request: We should provide a "submitting" spinner, and a quick note that says "A confirmation email is on its way!"

**Comment by ScottDowne** [comment]
I was poking around the firefox for mobile snippet code, and looks like it was able to handle signups via SMS?

Might be something we look into, or maybe look into for the future?

Also looks like we can do this directly from the snippet to exact target (basket) which is good new!

**Comment by ScottDowne** [comment]
I really like how the privacy policy is displayed here: ![2016-01-13_1206_-_jcollings_s_library](https://cloud.githubusercontent.com/assets/9830371/12306941/497c9a4e-b9f1-11e5-81a7-5fd97dc5a55c.png)

However, that only works if the policy is really short.

**Comment by brettgaylor** [comment]
I don't think text is describing the privacy policy - its describing the
CTA, which we can def keep short!

**Comment by ScottDowne** [comment]
Sounds logical to me.

I'm making the text easily configurable, but, I'm making the location and style based on the hope that the text is on the short side.

**Comment by ScottDowne** [comment]
For dev purpose I have been using a page that looks like this:

![snapshot1](https://cloud.githubusercontent.com/assets/197334/12402106/dccbfa64-bdf8-11e5-9d67-43f970b778f2.png)

I basically just stole things we have already used, and got a few tips from Ricardo. It's not final.

Test drop down is actually country...

**Comment by ScottDowne** [comment]
WIP: http://scottdowne.github.io/eoy-snippets/signup-snippet/

Has working signup, all bits of text is configurable, you're just seeing my placeholder text.

UI is not final.

I forgot to add a video, so I'll look into that tomorrow.

I also forgot the waiting spinner.

Thoughts?

**Comment by OpenMatt** [comment]
@valianttry @lovegushwa thoughts on above?

**Comment by valianttry** [comment]
This is the point where we need to:
-- check in with the snippet team (I already sent an email to Jean) to get reaction and feedback; and 
-- check with legal to confirm this configuration. 

I'll nudge Jean today.

andrea

**Comment by OpenMatt** [comment]
Flagging w. needs discussion just to ensure we're on track in today's meeting

**Comment by valianttry** [comment]
Just got feedback from Jean that we may  need to add email preference (text / HTML) and an optin checkbox. We are checking in with Marshall Erwin on specifics.

**Comment by ScottDowne** [comment]
@valianttry Question, is there a video somewhere yet?

**Comment by valianttry** [comment]
Nope, you can just use a standin from youtube or somewhere i suppose? Video shoot is happening this weekend.

**Comment by ScottDowne** [comment]
@valianttry hm, this might be a quesrtion for Jean.

I'll need to know what kind of restrictions snippets have on videos. I suspect youtube is a no go.

**Comment by ScottDowne** [comment]
@valianttry ok cool, heard back from Jean.

Video needs to be a .webm and needs to be 480x270.

For now I'll use something like that for the standin!

**Comment by ScottDowne** [comment]
I added the ability to load a video, it's optional.

Please ignore the fact that it's a random video...

http://scottdowne.github.io/eoy-snippets/signup-snippet/

**Comment by brettgaylor** [comment]
This is rad.

> On Jan 20, 2016, at 4:58 PM, ScottDowne notifications@github.com wrote:
> 
> I added the ability to load a video, it's optional.
> 
> Please ignore the fact that it's a random video...
> 
> http://scottdowne.github.io/eoy-snippets/signup-snippet/ http://scottdowne.github.io/eoy-snippets/signup-snippet/
> —
> Reply to this email directly or view it on GitHub https://github.com/MozillaFoundation/encryption_campaign/issues/23#issuecomment-173391632.

```
Brett Gaylor
Open Web Fellows + Advocacy Media
Mozilla






```

**Comment by OpenMatt** [comment]
nice work guys!
@valianttry what are the next steps here?

**Comment by valianttry** [comment]
We are paused on build until I get answers from Marshall Erwin / privacy / legal.

**Comment by valianttry** [comment]
Word from Marshall is we may need to add an optin checkbox, but he is checking with legal. Don't add it yet, this is just a heads up.

**Comment by OpenMatt** [comment]
- Want to make sure @ScottDowne 's next steps are clear on this this week. 
- Do folks think a kick-off meeting would be helpful to ensure we're clear on next steps for this build? 
  @valianttry @ScottDowne @lovegushwa

**Comment by valianttry** [comment]
We can't move forward until I get definitive word on required fields from legal. This is stuck for the moment at that step. No kickoff is needed ATM.

**Comment by valianttry** [comment]
Meeting now scheduled with snippet team and legal team for this Thurs 1/28

**Comment by OpenMatt** [comment]
Just to underline: this means we will be blocked on all snippet-related production work until Thursday minimum

**Comment by valianttry** [comment]
Some helpful info from Jean Collings:

"
-- If the main goal of this campaign is opt-ins, I prefer we push the video and the opt-in form together. Since videos are a big change on about:home, we prefer to keep this type of content to 3 days (to avoid user fatigue). More than happy to talk/think this one through with you.

-- If we are doing videos above the opt-in form, then Scott will need to create a template that allows for that (right now it's only the opt-in form template as far as I know). We have a "Simple Video over Search Bar" template - in which he can grab that code from Giorgos. Hope that helps!"

**Comment by OpenMatt** [comment]
- @valianttry @ScottDowne @lovegushwa what are the next steps here? 
- @ScottDowne are you working on this? Or still waiting for direction? 
- @lovegushwa does this feedback from Jean influence design?

**Comment by ScottDowne** [comment]
The video bits are implemented and optional. I'll be adding the checkbox
today when I get a chance.

**Comment by lovegushwa** [comment]
It shouldn't take much. I'll connect with @ScottDowne today.

**Comment by ScottDowne** [comment]
When do we plan to launch this snippet?

@OpenMatt @valianttry ^^

**Comment by valianttry** [comment]
@ScottDowne ideally it will be ready to go live on Feb 10 when our first video launches.

**Comment by valianttry** [comment]
Confirmed from legal we do need a checkbox next to this text:
- I’m okay with Mozilla handling my info as explained in this Privacy Notice. 
  (NOTE: The words "this Privacy Notice" should be linked to: https://www.mozilla.org/en-US/privacy/websites/.)
- Also confirmed we can now accept name in Basket!

So the snippet should have these elements:

[video]
[search bar]
"Mozilla stands up for online privacy. Will you join us? Sign up to get important email updates and opportunities to stand with us."
[first name]
[country]
[submit]
[checkbox] I’m okay with Mozilla handling my info as explained in this Privacy Notice.

After someone clicks "submit" we can show a short message that says:
"Great! Look for a confirmation email in your inbox to complete your subscription to our email list."

cc @shaghdoosti

**Comment by valianttry** [comment]
cc @kellymj @jessilyndavis

**Comment by ScottDowne** [comment]
@valianttry about 19 hours ago a patch landed in basket to allow first name and last name.

**Comment by valianttry** [comment]
@ScottDowne any update on build progress?

**Comment by ScottDowne** [comment]
@valianttry no, I pushed it back to this week when I found out it was Feb 5th and not Jan 29th in priority of other things launching on Jan 29th as well.

However, I started back on this today, and should have updates end of day.

Things I need to do still:
- [ ] Waiting spinner state.
- [ ] Checkbox
- [ ] Update copy in staging
- [ ] Implement any design changes/feedback.

I see this as a day or two max.

I'm also going to need a few days to review/land it in the snippet service, so if I can get the above done tonight, it would be ideal.

**Comment by ScottDowne** [comment]
@valianttry question, the Fed 5th milestone, is that dev is done and we're ready to ship, or is that when we actually turn on the snippet?

**Comment by valianttry** [comment]
@ScottDowne the snippet won't go live until Feb 10. So, the former.

**Comment by lovegushwa** [comment]
@ScottDowne we should build something like this:
<img width="1274" alt="screen shot 2016-02-01 at 10 28 10 pm" src="https://cloud.githubusercontent.com/assets/12228435/12741731/f61076ee-c933-11e5-8038-ba16900d9add.png">
(This option looks better)

or this:
<img width="1276" alt="screen shot 2016-02-01 at 10 28 34 pm" src="https://cloud.githubusercontent.com/assets/12228435/12741743/0e768fca-c934-11e5-8e04-0d163ab638ea.png">
(This option may be better from a UI perspective.)

Let's discuss at the 12pm (EST) kickoff meeting tomorrow.

**Comment by valianttry** [comment]
This looks great -- @lovegushwa if it helps the design, I think we could drop the highlighting on the "Sign up..." sentence. It's a really short call to action (unlike the fundraising snippets) so I don't think it will impact conversion much to lose it, but might help the design. We should specify "First Name" in the field label. That's because the purpose is so that we can address people by first name in email salutation. We want to greet you with "Hello Jordan,". As it is now, the label "name" might lead people to type both first and last, so resulting emails would say, "Hello Jordan Gushwa" which is oddly formal. I'm recommending we forego last name because it is a bit of PII (Personally Identifiable Information) that we actually have no use for in advocacy. If we need last name on an action, we can simply add it to a future form. But for email optin, we don't need it.

Also the privacy policy language must match my earlier comment exactly (to conform to recommendations from legal). That means "this privacy policy" should say "this Privacy Notice" instead, and all 3 words should be linked, and Privacy Notice should be capitalized.

**Comment by brettgaylor** [comment]
Can I just check that whatever we use for embedding the video can record number of plays # of users who played entire video?  Ideally we’d also get # of users who watched 1/2 of video.

> On Feb 2, 2016, at 1:04 AM, Andrea Wood notifications@github.com wrote:
> 
> This looks great -- @lovegushwa https://github.com/lovegushwa if it helps the design, I think we could drop the highlighting on the "Sign up..." sentence. It's a really short call to action (unlike the fundraising snippets) so I don't think it will impact conversion much to lose it, but might help the design. We should specify "First Name" in the field label. That's because the purpose is so that we can address people by first name in email salutation. We want to greet you with "Hello Jordan,". As it is now, the label "name" might lead people to type both first and last, so resulting emails would say, "Hello Jordan Gushwa" which is oddly formal. I'm recommending we forego last name because it is a bit of PII (Personally Identifiable Information) that we actually have no use for in advocacy. If we need last name on an action, we can simply add it to a future form. But for email optin, we don't need it.
> 
> Also the privacy policy language must match my earlier comment exactly (to conform to recommendations from legal). That means "this privacy policy" should say "this Privacy Notice" instead, and all 3 words should be linked, and Privacy Notice should be capitalized.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/MozillaFoundation/encryption_campaign/issues/23#issuecomment-178413658.

```
Brett Gaylor
Open Web Fellows + Advocacy Media
Mozilla






```

**Comment by jessilyndavis** [comment]
Heyo - my quick 2 cents: I know having a name is important for ongoing fostering of relationships with these new email subscribers, but I'm curious how that extra field impacts conversion. Would it be worth it to a/b test a form with the name field in tact - and one with just email address?  

Reason I ask:
If name is a good practice, but doesn't significantly impact ongoing engagement with the individual vs. adding in the name field does significantly impact the number of new signups you get - it might be worth it to leave off the name field.

**Comment by ScottDowne** [comment]
@jessilyndavis I can make the name field optional for us, and we can configure it as we want, or spin up two and configure them differently to a/b test.

@brettgaylor There is some moco level analytics in the snippet I think I can tap into, I'll add that to list of things I need to look into.

@valianttry The highlight is already optional, so yes, we can easily turn it off and on as we want. I agree, first name is better.

@lovegushwa Not sure how much test time we'll get, but possibly we do both versions?

**Comment by ScottDowne** [comment]
@lovegushwa Here is the screenshot from out experiments: 
![snippets](https://cloud.githubusercontent.com/assets/197334/12760657/64dea326-c9b5-11e5-8959-98152c9a995f.png)

**Comment by valianttry** [comment]
@jessilyndavis we discussed this tradeoff .... @ScottDowne is building it so we could test... Here's some of our thinking:
- Since the video is only on the snippet for 2-3 days and we're racing the clock, I opted to simplify things with less optimization testing on the snippet itself. But we could test name field variants during the course of the campaign if we want.
- More importantly, our primary KPI for the encryption campaign is not acquisition, it is long-term conversion on our campaign emails, so with that logic using real first name in email salutations over time almost always improves conversion. 
- If / when we do test, in addition to simple acquisition conversion (whether name reduces optin %) -- we would also want to analyze whether the primary KPI is impacted. in other words, does no first name reduce conversion on future actions vs. having the first name in email salutation?

**Comment by lovegushwa** [comment]
@ScottDowne here are the final options.
CC @valianttry 

<img width="1274" alt="screen shot 2016-02-02 at 12 48 21 pm" src="https://cloud.githubusercontent.com/assets/12228435/12763836/7a2f2df4-c9ab-11e5-8ee0-ecdab18774a9.png">

<img width="1273" alt="screen shot 2016-02-02 at 12 48 37 pm" src="https://cloud.githubusercontent.com/assets/12228435/12763839/8012372a-c9ab-11e5-9974-5e047a390dd6.png">

<img width="1274" alt="screen shot 2016-02-02 at 12 48 52 pm" src="https://cloud.githubusercontent.com/assets/12228435/12763842/86ea4e3e-c9ab-11e5-87e2-f2205925281c.png">
**you can also put the spinner in the center of this gray box**

**Comment by ScottDowne** [comment]
@lovegushwa One thought, can I put the spinner in the button itself? Example: 
![spin](https://cloud.githubusercontent.com/assets/197334/12763934/21db94f2-c9c5-11e5-89a7-d3a188ed5faa.png)

At the time of displaying the spinner, we don't know if it'll pass or fail yet. If it passes, we hide the UI and display success message. If it fails, we don't.

**Comment by valianttry** [comment]
@ScottDowne we didn't really talk about what would be required. I wonder if @jessilyndavis has any thoughts? My instinct says that only email + privacy checkbox should be required, country and name should not be (to honor user choice). Does that throw a wrench in the works?

**Comment by jessilyndavis** [comment]
+1
email + privacy checkbox should be required
country and name should not be required

would be worth asking Brian/Marshall just to make sure there's not a legal reason. (I dont think so, it's more a nice to have as far as I know)

**Comment by valianttry** [comment]
@jessilyndavis Brian commented in his email that our existing optin configurations were good on the site, and they only require email + checkbox so I think we're OK. I'd definitely want to check with them if we were thinking about requiring new PII to optin, but we're not.

**Comment by ScottDowne** [comment]
OK cool, so name and privacy required, rest is optional.

**Comment by lovegushwa** [comment]
spinner in the button is cool.

**Comment by ScottDowne** [comment]
@valianttry Any legal concerns regarding the error message for when the privacy checkbox is not selected when they submit?

**Comment by valianttry** [comment]
@ScottDowne let's use what is currently on the homepage signup:

"Please check this box if you want to proceed."

![screen_shot_2016-02-02_at_1_19_26_pm](https://cloud.githubusercontent.com/assets/9830371/12764615/afbe3aa6-c9af-11e5-9693-06f6a9348f4f.png)

**Comment by OpenMatt** [comment]
@ScottDowne you all good to proceed on this one? Got what you need?

**Comment by ScottDowne** [comment]
@OpenMatt Yup, so far so good. Should be done today.

@lovegushwa for the placement of the play triangle in the button, I'm finding some places center it literally and some center it visually. Which of these would you consider "centered":
![center](https://cloud.githubusercontent.com/assets/197334/12788839/455baf8a-ca69-11e5-88aa-e1c97f777d90.png)
![center1](https://cloud.githubusercontent.com/assets/197334/12788841/47f02a8c-ca69-11e5-9696-3c0e09db62fd.png)

 Another way to look at it is this, bottom one is centered horizontally, top one is centered via the same distance from each corner to the closest button edge.

**Comment by lovegushwa** [comment]
@ScottDowne visually centered! (top button)

**Comment by ScottDowne** [comment]
@lovegushwa thanks! I would agree!

**Comment by chelseanovak** [comment]
@valianttry I was just chatting with the Snippets team and they had two questions: 

1) Is "Mozilla stands up for online privacy. Will you join us? Sign up to get important email updates and opportunities to stand with us." the final snippet copy? If not, we need the final copy by the end of this week. 

2) Is the snippet copy built into the the embedded form or does the copy go into the usual snippet service? This could also be a question for @ScottDowne

**Comment by ScottDowne** [comment]
@chelseanovak It goes into the usual snippet service :)

I'm making it as flexible and configurable as I can.

**Comment by ScottDowne** [comment]
OK, this is what I have done up to this point.

http://scottdowne.github.io/eoy-snippets/signup-snippet/

It's done from what feedback and requests I was able to catch. If I missed something, let me know.

@brettgaylor I setup metrics for video play, finish, half way, and replay. It was easy, so I figured why not track replay clicks too! It's also tracking signups.

**Comment by valianttry** [comment]
That is the final snippet copy @chelseanovak

**Comment by valianttry** [comment]
@ScottDowne the checkbox error message says:

![about_home_snippet_preview](https://cloud.githubusercontent.com/assets/9830371/12805963/4c6f1170-cab6-11e5-97f6-3de2c7cdfcae.png)

But the example text from the original form at mozilla.org says:
"Please check this box if you want to proceed." 

Did you or @lovegushwa make this change? Is it because the box doesn't "glow" if left unchecked like on the mozilla.org form? I recommend using the original text, I still think it works and it's approved by legal.

**Comment by ScottDowne** [comment]
@valianttry Right, missed that. Meant to go back to this.

The issue is I couldn't get the dialog with the arrow to pop up, after about an hour or so fiddling with it, and at the time didn't feel worth it when I still had other pressing things to do. Now that everything else is done, I can go back to it.

Interestingly, that error dialog is a default browser dialog, and the text "Please check this box if you want to proceed." is also default error text for generic checkboxes. I wonder if this isn't actually a legal chosen wording if it's built into the browser. It's also possible legal has approved the default text, which would make sense.

I think it's worth it for me to go back and try the built in dialogs.

**Comment by valianttry** [comment]
@ScottDowne only if you have time -- i wouldn't sweat it too much.

**Comment by ScottDowne** [comment]
@valianttry Here it is with the native validation: http://scottdowne.github.io/eoy-snippets/signup-snippet/

It's quite a bit less code. I kinda like it this way :)

**Comment by ScottDowne** [comment]
One interesting note here is it only shows you one error message at a time, top most input first.

**Comment by chelseanovak** [comment]
@valianttry Excellent. Two questions from the IPC team:

1) Has this been run past brand or legal? We ask because in November we were specifically directed to state that we stand for choice vs for privacy itself. It may be because of all the additional caution we were taking with Privacy Browsing with Tracking Protection and, hopefully, doesn't apply here. We'd rather the copy remain unchanged, but just wanted to flag. 

2) Will this run with a video asset as well? Like this?

[EMBEDDED VIDEO]

Mozilla stands up for online privacy. Will you join us? Sign up to get important email updates and opportunities to stand with us

[FORM]

Thanks!
C

**Comment by valianttry** [comment]
@chelseanovak you can see the actual snippet in staging here:
http://scottdowne.github.io/eoy-snippets/signup-snippet/

I am meeting with legal today to go over the campaign. 

Andrea

**Comment by valianttry** [comment]
@OpenMatt is @alanmoo or @alicoding going to take this over the finish line? We're so close! :checkered_flag:

**Comment by ScottDowne** [comment]
@valianttry template landed this morning, I think next step is to loop in with Jessica to get the variables the way we want them on staging.
